perfkitbenchmarker/benchmarks/cassandra_stress_benchmark.py

```diff
@@ -174,100 +304,151 @@ def Prepare(benchmark_spec):
+                           cassandra_stress_profile_ops='insert=1',
+                           cassandra_stress_pop_size=None,
+                           cassandra_stress_pop_dist=None,
+                           cassandra_stress_pop_params=None):
+  """Start all loader nodes as Cassandra clients and run stress test.
 
   Args:
     benchmark_spec: The benchmark specification. Contains all data
         that is required to run the benchmark.
+    num_keys: The number of operations cassandra-stress clients should issue.
+    cassandra_stress_command: The cassandra-stress command to use.
+    cassandra_stress_profile_ops: The ops to use with user mode.
+    cassandra_stress_pop_size: The population size.
+    cassandra_stress_pop_dist: The population distribution.
+    cassandra_stress_pop_params: String representing additional
+        population parameters.
   """
-  loader_vms = benchmark_spec.vm_groups[CLIENT_GROUP]
-  num_loaders = len(loader_vms)
-  cassandra_vms = benchmark_spec.vm_groups[CASSANDRA_GROUP]
-  data_node_ips = [vm.internal_ip for vm in cassandra_vms]
-  if not FLAGS.num_keys:
-    FLAGS.num_keys = NUM_KEYS_PER_CORE * cassandra_vms[0].num_cpus
-    logging.info(
-        'Num keys not set, using %s in cassandra-stress test.',
-        FLAGS.num_keys)
-  keys_per_vm = FLAGS.num_keys / num_loaders
-  if FLAGS.num_keys % num_loaders:
-    logging.warn(
-        'Total number of keys rounded to %s (%s keys per loader vm).',
-        keys_per_vm * num_loaders, keys_per_vm)
-  logging.info('Executing the benchmark.')
-  args = [((loader_vms[i], i, keys_per_vm, data_node_ips), {})
-          for i in xrange(0, num_loaders)]
-  vm_util.RunThreaded(RunTestOnLoader, args)
-
-
-def WaitForLoaderToFinish(vm):
-  """Watch loader node and wait for it to finish test.
-
-  Args:
-    vm: The target vm.
-  """
-  result_path = _ResultFilePath(vm)
-  while True:
-    resp, _ = vm.RemoteCommand('tail -n 1 ' + result_path)
-    if re.findall(r'END', resp):
-      break
-    if re.findall(r'FAILURE', resp):
-      vm.PullFile(vm_util.GetTempDir(), result_path)
-      raise errors.Benchmarks.RunError(
-          'cassandra-stress tool failed, check %s for details.'
-          % posixpath.join(vm_util.GetTempDir(),
-                           os.path.basename(result_path)))
-    time.sleep(SLEEP_BETWEEN_CHECK_IN_SECONDS)
+  try:
+    loader_vms = benchmark_spec.vm_groups[CLIENT_GROUP]
+    num_loaders = len(loader_vms)
+    cassandra_vms = benchmark_spec.vm_groups[CASSANDRA_GROUP]
+    data_node_ips = [vm.internal_ip for vm in cassandra_vms]
+    cassandra_stress_pop_size = cassandra_stress_pop_size or num_ops
+    ops_per_vm = num_ops / num_loaders
+    pop_per_vm = cassandra_stress_pop_size / num_loaders
+    if num_ops % num_loaders:
+      logging.warn(
+          'Total number of ops rounded to %s (%s ops per loader vm).',
+          ops_per_vm * num_loaders, ops_per_vm)
+      logging.info('Executing the benchmark.')
```

Comment:
Is this indentation correct (only log that you are executing if `num_ops % num_loaders` is truthy)?